File: ProjectTracker20cs/PTWin/StatusBusy.cs
using System;
using System.Windows.Forms;

namespace PTWin
{
  public class StatusBusy : IDisposable
  {
    public StatusBusy(string statusText)
    {
      MainForm.Instance.StatusLabel.Text = statusText;
      MainForm.Instance.Cursor = Cursors.WaitCursor;
    }

    // IDisposable
    private bool _disposedValue = false; // To detect redundant calls

    protected void Dispose(bool disposing)
    {
      if (!_disposedValue)
        if (disposing)
        {
          MainForm.Instance.StatusLabel.Text = string.Empty;
          MainForm.Instance.Cursor = Cursors.Default;
        }
      _disposedValue = true;
    }

    public void Dispose()
    {
      // Do not change this code. Put cleanup code in Dispose(bool disposing) above
      Dispose(true);
      GC.SuppressFinalize(this);
    }
  }
}

File: ProjectTrackervb/PTWin/StatusBusy.vb
Public Class StatusBusy

  Implements IDisposable

  Private mOldStatus As String
  Private mOldCursor As Cursor

  Public Sub New(ByVal statusText As String)

    mOldStatus = MainForm.StatusLabel.Text
    MainForm.StatusLabel.Text = statusText
    mOldCursor = MainForm.Cursor
    MainForm.Cursor = Cursors.WaitCursor

  End Sub

  ' IDisposable
  Private disposedValue As Boolean = False ' To detect redundant calls

  Protected Overridable Sub Dispose(ByVal disposing As Boolean)
    If Not Me.disposedValue Then
      If disposing Then
        MainForm.StatusLabel.Text = mOldStatus
        MainForm.Cursor = mOldCursor
      End If
    End If
    Me.disposedValue = True
  End Sub

  Public Sub Dispose() Implements IDisposable.Dispose
    ' Do not change this code.  
    ' Put cleanup code in Dispose(ByVal disposing As Boolean) above.
    Dispose(True)
    GC.SuppressFinalize(Me)
  End Sub

End Class
